Big upgrade and merge of PR

mermetbt opened the pull request:
Hello,

Thanks for the work done on TimeOff.Management!

Here is the summary of changes:

- Slack Support (Merge PR #263)
- Google Auth provider
- Anniversary Calendar
- Upgrade Express to 4.x, Sequelize to 5.x (merge PR #345, and PR #346)
- Upgrade to NodeJS 16 (issue #528)
- Refactoring companyId to company_id
- LDAP search filter (merge #359 for issue #352)
- Fix error overlapping booking when using mysql db (merge #396, issue #249)
- New Docker build and docker-compose (merge PR #567)
- Add Prettier and ESLint to standardize the coding style and identify issues
- Employee comments (merge #306, issue #133)

Theses are breaking changes for current version running on production, I would more consider this as a starting point for a v2.

The code is not stable yet, and I'm still working at fixing remaining issues.

Commit:
Added Slack support. New class and new templates, both based off the original email ones. Modified calls for emails to add the Slack calls.

Commit:
Removed any specific development data. Added instructions to settings. Added unit test. Moved settings to proper place.

Commit:
Merge branch 'slack'

Merging Slack integration.

Commit:
Correct config file. Remove unecessary comments and console.logs. Restore proper footer. Restore deleted template.

Commit:
Restore app config original parameters. Clean up code on email.js file. Clean up CSS files. Restore markup on email template.

Commit:
Restore original fake email address on app config. Restoring spacing and remove console.log on login.js lib file.

Commit:
Removed the DB migration and instead added the new slack_username field to the database schema using the correct tool. This avoids any possible issues for integration with current master and anyone installing the app from scracth.

Commit:
Correct location of promise_reset_password_slack call.

Commit:
Add slack_username to the index vector correctly on the CSV importer.

Commit:
Set 'send_slacks' config flag to false as default.

Commit:
Changed slack_username column in Users to allow nulls to facilitate imports. Correct sending msg to supervisor twice instead of supervisor and requester on new leave request. Fixed issue on same method on finishing promise.

Commit:
Fixed and added missing titles to all the pages.

simllll committed:
add google auth provider

simllll committed:
cleanup

simllll committed:
check for mail

simllll committed:
only need mail

simllll committed:
use prepare for session

simllll committed:
add some logging and fix domain check

simllll committed:
library update and some rewrites

simllll committed:
let's how the automatic tests perform...

simllll committed:
leave request check for user

simllll committed:
leave request check for user

simllll committed:
smtp

simllll committed:
smtp

simllll committed:
email?=

simllll committed:
email?=

simllll committed:
email?=

simllll committed:
email?=

simllll committed:
email?=

simllll committed:
email?=

simllll committed:
asnyc mails

simllll committed:
fix merge

simllll committed:
merged title and slack PR (by Morthor)

simllll committed:
rm logs

simllll committed:
revert calendar

simllll committed:
simple anniversary calendar: added json output as comment in ical format.
fixed upload/restore function by using "multer" instead.

simllll committed:
mysql2 as optional dep

simllll committed:
sqlite3 to optional dep

simllll committed:
add Anniversaries calendar link

simllll committed:
format all js files with prettier

simllll committed:
[WIP] upgrade express to 4.x, sequelize to 5.x,.. and more

simllll committed:
[WIP] refactoring companyId -> company_id etc to fix errors and mixups

simllll committed:
fix: user find of underscore has no findOne

simllll committed:
fix/revert: active user filter / use null again

simllll committed:
fix: department validation

simllll committed:
feat: add migration for copmanyId and DepartmentId

rafacouto committed:
Configurable LDAP search filter. Issue #352

rafacouto committed:
Validation: ldap_config.searchfilter

rafacouto committed:
LDAP search filter default value.

Larykath committed:
#249 fixes the error "overlapping booking" when using mysql db

simllll committed:
fix: remove slack required username

simllll committed:
feat: add leave type id

simllll committed:
feat: encode base 64 again

simllll committed:
feat: encode base 64 again

dougbreaux committed:
minimum changes for docker build to succeed

mermetbt committed:
Merge pull request #1 from dougbreaux/master

minimum changes for docker build to succeed

mermetbt committed:
Merge pull request #2 from dkraemerwork/feature/#249_mysql_overlapping_booking_fix

#249 fixes the error "overlapping booking" when using mysql db

mermetbt committed:
Merge branch 'ldap-filter'

mermetbt committed:
Merge prettier.

mermetbt committed:
Add package-lock and editorconfig.

mermetbt committed:
Add redis dependency.

mermetbt committed:
Update package-lock.

mermetbt committed:
Add docker compose. Fix merge issue.

mermetbt committed:
Fix invalid version.

mermetbt committed:
Fix.

mermetbt committed:
Fix Dockerfile.

mermetbt committed:
Change Dockerfile.

mermetbt committed:
Add support for env.

mermetbt committed:
Fix issues.

mermetbt committed:
Prettier.

mermetbt committed:
Prettier.

mermetbt committed:
Add prettierrc.

mermetbt committed:
Set quote to double quote.

mermetbt committed:
Add semi columns.

mermetbt committed:
Merge.

mermetbt committed:
Prettier.

mermetbt committed:
Fix.

mermetbt committed:
Fix issues.

mermetbt committed:
Add eslint.

mermetbt committed:
Prettier.

mermetbt committed:
Eslint

mermetbt committed:
Eslint and prettier.

mermetbt committed:
Set the test database to sqlite.

mermetbt committed:
Fix.

mermetbt committed:
Add environment variables.

mermetbt committed:
fix: Solve part of the issues for the tests to pass.

mermetbt committed:
fix: Bug fixes and force consistency in the database.

mermetbt committed:
fix: Issues with leave types and request approval.

mermetbt committed:
fix: Fix leave types issue.

mermetbt committed:
fix: Improve consistency of the database.

mermetbt committed:
Create docker-image.yml

mermetbt committed:
fix: Email from not set in emails.

mermetbt committed:
Merge branch 'master' of github.com:AlienGen/timeoff-management-application

mermetbt committed:
ci: keep bin dir into the docker image.

mermetbt committed:
fix: LDAP configuration issue when checkboxes are not ticked. LDAP settings not saved when using MySQL (datatype issue).

mermetbt committed:
fix: Some environment variables were not working.

mermetbt committed:
fix: Change datatypes for comments and timezone.

mermetbt committed:
feat: Change configurations and update the documentation.

mermetbt committed:
fix: Google auth domains.

mermetbt committed:
fix: Settings company wide schedule.

mermetbt committed:
fix: Change table names for comment and schedule.

mermetbt committed:
fix: Checkboxes in department settings not working.

spinriko commented:
merging in aliengen fork

alexleach commented:
> The code is not stable yet, and I'm still working at fixing remaining issues.

Would you say this is still the case?

mermetbt commented:
@alexleach We are currently using it on "production" with the corresponding [Docker image](https://hub.docker.com/r/aliengen/timeoff-management-application) we built.

It is working for us with our current usage. I would say the main issue we have is that the variables in the email template are not set properly. 

Unfortunately, I do not have availability to fix more issues. I'm staying available if anyone want to contribute and make some progress with it.

alexleach commented:
Thanks for the quick reply! Your docker image looks great, lots of options available! It is almost two years old though, so I imagine lots of packages need bumping?

I spent some time yesterday working on building a docker image from the main repo. I kept having issues with node-gyp trying to build an outdated sqlite3 extension, and then I think issues with the preinstall script. Plus there are 12 Critical Security issues that require breaking package updates... This main repo has been sadly neglected, unfortunately.

Thanks for all your team's efforts on trying to keep this project alive. I'll try again to get it deployed internally later today. I've got it to a state where my Dockerfile and compose.yaml files can start the node app, redis and mariadb images now, so I think I'm nearly there...

If I can contribute anything obvious, I'll submit a PR, but I wasn't intending on putting this much time into it tbh!

mermetbt committed:
Merge branch 'timeoff-management:master' into master

mermetbt committed:
fix: Issue with smtp without TLS. Issue with the mail template variables not passing. Update of dependencies. Switch to node 20.

mermetbt committed:
fix: Add missing dependency in Dockerfile.

mermetbt committed:
fix: SMTP configuration via env variable with STARTTLS.

mermetbt commented:
I've updated node to the version 20 and fixed a few issues including the issue with the mail template variables.

There is still a need to upgrade all of the dependencies, but this is another amount of work...

essys committed:
Fix boolean conversion error in departments.js

This fixes a TypeError occurring when adding a department:

    TypeError: Expected a string but received a Boolean

The issue was in `departments.js` where `validator.toBoolean()` was called on a boolean expression. The `validator` library expects a string input, so passing a boolean caused the error.

Changes
- Replaced:

```js
const include_public_holidays = validator.toBoolean(
    getParam('include_public_holidays') === 'on'
)
const is_accrued_allowance = validator.toBoolean(
    getParam('is_accrued_allowance') === 'on'
)

mermetbt committed:
Merge pull request #14 from essys/fix/boolean-to-string-department

Fix boolean conversion error in departments.js
